rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```
@@ -849,13 +850,23 @@ object ProcNormalizeMatcher {
                     case lbi: RepeatedBindImpl =>
                       (lbi.listname_.toList, lbi.name_, lbi.nameremainder_).pure[M]
                   }
-                  .map(x => (x, true))
+                  .map(x => (x, true, false))
+            }
+          case rl: ReceiptPeek =>
+            rl.receiptpeekimpl_ match {
+              case ls: PeekSimple =>
+                ls.listpeekbind_.toList
+                  .traverse {
+                    case lbi: PeekBindImpl =>
```

can you use a total function here?

---

No

---

so if the case doesn't match we get a match failure... not nice

---

we should raise an error if the receipt type matches none of these.